### ISSUE TITLE
URL encoding

### ISSUE THREAD

**Issue by DonKarlssonSan** [keesschollaart81/vsts-arm-outputs#76]
We use ARM Outputs in several Azure DevOps Release Pipelines. 
This week we've noticed that our trigger URLs for Logic Apps get "double encoded".

Expected output:
`https://prod-57.westeurope.logic.azure.com:443/workflows/GUID/triggers/manual/paths/invoke?api-version=2016-06-01&sp=%2Ftriggers%2Fmanual%2Frun&sv=1.0&sig=CODE
`

Actual output:
`https://prod-57.westeurope.logic.azure.com:443/workflows/GUID/triggers/manual/paths/invoke?api-version=2016-06-01&sp=%252Ftriggers%252Fmanual%252Frun&sv=1.0&sig=CODE
`

As you can see `%2F` becomes `%252F` 

I noticed that there was a 5.3 release made just recently. I guess we get that automatically because we use Task version 5.*

Can it be a bug introduced in the latest release?

**Comment by AlexandreColnec** [comment]
I have the same issue.
Change the version to 4.* seems to work if your agent is not a unix.

**Comment by JonnyMattsson** [comment]
I have the same issue. The connectionstring to a sql database that gets a '%25' inserted in the middle of the password. Changed the version to 4.* and that works.

**Comment by raviteja333** [comment]
Yes, looks like changing to 4.* worked. But, it was giving a deprecated usage warning.

**Comment by praveenvinay** [comment]
Same issue here.  Using 4.* is comparatively slower and deprecated.

**Comment by nspiteri** [comment]
In version 6 the problem seems to still exist.  However as mentioned earlier, version 4 is ok.

**Comment by bahadir-be** [comment]
Agree!!
Changed the version to 4.* seems to work with warnings.

**Comment by cniak** [comment]
I see it's more internal change in Azure DevOps code than changes in the vsts-arm-outputs task.

The fix will be automatically used in March 2021.
https://github.com/microsoft/azure-pipelines-agent/blob/master/docs/design/percentEncoding.md

Conclusion:
now you can go back to ARM Output v6, but first you need to apply variable DECODE_PERCENTS = true in your pipelines.

**Comment by bahadir-be** [comment]
> 
> 
> I see the fix will be gone automatically in March 2021.
> https://github.com/microsoft/azure-pipelines-agent/blob/master/docs/design/percentEncoding.md
> 
> so you can use the newest (faster version) of ARM Output, but first you need to define variable mentioned in the link above.

Thanks cniak,
My problem is replaced **''%25''** instead **''%'**' on SAS token key.

ex :
orjinal output SAS Token value :
sv=2015-04-05&ss=b&srt=o&sp=w&se=2022-05-02T00%3A00%3A00.0000000Z&sig=XXXXXXTqrROM7gU4uAeVs9PC5yO8KR2kSi9z5rO1qBs%3D_

ARM Outputs V6 's value : 
sv=2015-04-05&ss=b&srt=o&sp=w&se=2022-05-02T00%253A00%253A00.0000000Z&sig=XXXXXXTqrROM7gU4uAeVs9PC5yO8KR2kSi9z5rO1qBs%253D_


So what exactly should my Powershell code be after or before ARM output job?

**Comment by bahadir-be** [comment]
> 
> 
> I see it's more internal change in Azure DevOps code than changes in the vsts-arm-outputs task.
> 
> The fix will be automatically used in March 2021.
> https://github.com/microsoft/azure-pipelines-agent/blob/master/docs/design/percentEncoding.md
> 
> Conclusion:
> now you can go back to ARM Output v6, but first you need to apply variable DECODE_PERCENTS = true in your pipelines.

Thanks for your feedback!
It's working after added above variable..
